lib/http/request.js

```diff
@@ -135,6 +135,18 @@ const makeRequest = async function (method, path, options, hooks) {
 
     response.payload = await Read(response, options.readOptions)
 
+    if (hooks.onPostRead) {
+      response = await hooks.onPostRead(options.id, response) || response
+    }
+    // because of the retry, the code is not going to hit the read step
```

Comment:
This comment needs more context, specifically about why a retry would be occurring. I'm also still not clear on why the payload wouldn't be set.

Reply:
Hi Tucker - for the retry that gets triggered, the options are set `read === false`, so the retry step doesn't hit the read which means the payload is not set. And because the retry is triggered after the read, it resumes after the read as well and the returned object from the retry was not read which means that we have to read it at the point otherwise, the returned response does not have a payload. 

1. send request that fails auth, payload was read to check that
2. triggers retry from hooks.onPostRead
3. retry request succeeds but is not read because options.read is set to false for retry
4. code continues from where original request left off at hooks.onPostRead, but since the retry request was successful and not read, it needs to get read now

Reply:
I am realizing that maybe I need to rethink this processing flow. We might need to set the retry to read = true.